node/pkg/governor/governor_monitoring.go

```diff
@@ -601,3 +602,21 @@ func (gov *ChainGovernor) publishStatus(hb *gossipv1.Heartbeat, sendC chan<- []b
 
 	sendC <- b
 }
+
+func (gov *ChainGovernor) IsTransactionEnqueued(emitterChainId vaa.ChainID, txHash ethCommon.Hash) bool {
+	gov.mutex.Lock()
+	defer gov.mutex.Unlock()
+
+	ce, exists := gov.chains[emitterChainId]
+	if !exists {
+		return false
+	}
+
+	for _, pe := range ce.pending {
+		if bytes.Equal(pe.dbData.Msg.TxHash.Bytes(), txHash.Bytes()) {
```

Comment:
The TxHash does not uniquely identify a wormhole message because one transaction could create multiple wormhole messages.